packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/log_stream_provider.dart

```diff
@@ -36,15 +40,22 @@ class DefaultCloudWatchLogStreamProvider
   final String _logGroupName;
   final CloudWatchLogsClient _client;
   static final DateFormat _dateFormat = DateFormat('yyyy-MM-dd');
+  final _cachedLogStreams = <String>{};
 
-  /// Creates CloudWatch log stream if does not exists and returns
-  /// the log stream name.
-  ///
-  /// Throws an [Exception] if fails to create the log stream.
   @override
-  Future<String> get logStream async {
+  FutureOr<String> get logStream async {
```

Comment:
Making this `FutureOr` is a decent idea, but by marking this `async` it will always return a `Future`.

In the cached case, this is the same as:

```dart
Future<String> get logStream {
   if (_cachedLogStreams.contains(logStreamName)) {
      return Future.value(logStreamName);
    }
}
```

Further, if you await the return of a `FutureOr` it is implicitly wrapped in `Future.value` and you cannot use the result synchronously.

That is to say, to fully leverage `FutureOr`, you need to switch on the return and treat it as a sealed class over `T` and `Future<T>`.

https://dartpad.dev/?id=dc5c257693d0d454c819ffcf38b626f0

Reply:
updated to Future<String>